crates/base-db/src/input.rs

```diff
@@ -324,6 +328,99 @@ pub struct CrateData {
     pub channel: Option<ReleaseChannel>,
 }
 
+impl CrateData {
+    /**
+    Check if [`other`] is almost equal to [`self`].
+    This method has some obscure bits. These are mostly there to be compliant with
+    some patches. References to the patches are given.
+    */
+    pub fn almost_eq(&self, other: &CrateData) -> bool {
```

Comment:
Let's call this `eq_ignoring_origin`, that's a bit more clear on what we ignore here in terms of equality